.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,26 @@
+name: deploy
+on:
+  workflow_run:
+    workflows: ["build-test"]
+    types:
+      - completed
+jobs:
+  deploy:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: [self-hosted, linux, ARM64]
+    env:
+      IMAGE_NAME: node_express_api
+      APP_SERVER_PORT: 3000
+      COMMIT_SHA_TAG: ${{ github.sha }}
```

Comment:
I don't think this will actually work, because this workflow is triggered off of the other workflow and not triggered by any git event. So `github.sha` will be null or something when used. And I'm not sure how to pass environment variables or parameters from the previous workflow (at least not in a clean simple way).

2 separate workflows isn't really necessary: this deploy job could just be added at the end of the first workflow and then the `github.sha` values would be accessible. But I wanted to test running multiple workflows and I think it's good practice to conceptually separate the process of building an image from the process of container deployment.